crates/trie/src/trie_cursor/subnode.rs

```diff
@@ -58,12 +55,11 @@ impl CursorSubNode {
     /// Creates a new `CursorSubNode` from a key and an optional node.
     pub fn new(key: Nibbles, node: Option<BranchNodeCompact>) -> Self {
         // Find the first nibble that is set in the state mask of the node.
-        let nibble = match &node {
-            Some(n) if n.root_hash.is_none() => {
-                CHILD_INDEX_RANGE.clone().find(|i| n.state_mask.is_bit_set(*i)).unwrap() as i8
-            }
-            _ => -1,
-        };
+        let nibble = node
+            .as_ref()
+            .filter(|n| n.root_hash.is_none())
+            .and_then(|n| CHILD_INDEX_RANGE.clone().find(|i| n.state_mask.is_bit_set(*i)))
```

Comment:
preserve panic on no state bit set
```suggestion
            .map(|n| CHILD_INDEX_RANGE.clone().find(|i| n.state_mask.is_bit_set(*i)).unwrap())
```